products/views.py

```diff
@@ -66,6 +66,55 @@ def get(self, request, product_id):
                 "features"              : [feature.name for feature in feature_category.feature_set.filter(product=product)]
             } for feature_category in set(FeatureCategory.objects.filter(feature__in=product.feature.all()))
         ]
+
         results["product_features"] = feature_result
 
-        return JsonResponse({'result':results}, status=200)
\ No newline at end of file
+        return JsonResponse({'result':results}, status=200)
+
+class PopularProduct(View):
+    def get(self, request):
+        products = Product.objects.all().order_by('-count')[:5]
+
+        total_results = []
+
+        for product in products:
+            category           = product.category
+            menu               = category.menu
+            ingredients        = product.ingredient.all()
+            product_selections = ProductSelection.objects.filter(product_id=product.id)
+
+            feature_result = [
+                {
+                    "feature_category_name" : feature_category.name,
+                    "features"              : [feature.name for feature in feature_category.feature_set.filter(product=product)]
+                } for feature_category in set(FeatureCategory.objects.filter(feature__in=product.feature.all()))
+            ]
+
+            results = [
+                {
+                    "menu_name"                  : menu.name,
+                    "menu_id"                    : menu.id,
+                    "category_name"              : category.name,
+                    "category_id"                : category.id,
+                    "category_description_title" : category.description_title,
+                    "category_description"       : category.description,
+                    "product_name"               : product.name,
+                    "product_id"                 : product.id,
+                    "product_description"        : product.description,
+                    "product_features"           : feature_result,
+                    "product_content"            : product.content,
+                    "product_content_image_url"  : product.content_image_url,
+                    "product_ingredients"        : [ingredient.name for ingredient in ingredients],
```

Comment:
```suggestion
                    "product_ingredients"        : [ingredient.name for ingredient in product.ingredient.all()],
```
계속 사용되는 변수가 아니므로 합치는 것이 어떨까 합니다.

Reply:
오 훨씬 보기 좋은것 같습니다!
product_selections 부분도 product.productselection_set.all()로 바꿔봤습니당 혹시 괜찮을까요?
수정했습니다 감사합니당!!! 🥳